app/helpers/template_helpers.py

```diff
@@ -14,291 +14,343 @@
+class CensusNISRAContextOptions(CensusContextOptions):
+    page_header_logo: str = "nisra-logo-en"
+    page_header_logo_alt: str = lazy_gettext(
+        "Northern Ireland Statistics and Research Agency logo"
     )
-    sign_out_url = url_for("session.get_sign_out")
-    static_content_urls = get_static_content_urls(language_code, base_url, theme)
-    footer_context = get_footer_context(
-        language_code,
-        static_content_urls,
-        sign_out_url,
-        theme,
+    header_logo: str = "nisra"
+    mobile_logo: str = "nisra-logo-en-mobile"
+    copyright_declaration: str = lazy_gettext(
+        "Crown copyright and database rights 2021 NIMA MOU577.501."
+    )
+    copyright_text: str = lazy_gettext(
+        "Use of address data is subject to the terms and conditions."
+    )
+    footer_links: Iterable[Mapping] = field(
+        default_factory=lambda: [
+            Link(
+                lazy_gettext("Help"),
+                "help/help-with-the-questions/online-questions-help/",
+            ).__dict__,
+            Link(lazy_gettext("Contact us"), "contact-us/").__dict__,
+        ]
+    )
+    footer_legal_links: Iterable[Mapping] = field(
+        default_factory=lambda: [
+            Link(lazy_gettext("Cookies"), "cookies/").__dict__,
+            Link(
+                lazy_gettext("Accessibility statement"),
+                "accessibility-statement/",
+            ).__dict__,
+            Link(
+                lazy_gettext("Privacy and data protection"),
+                "privacy-and-data-protection/",
+            ).__dict__,
+            Link(
+                lazy_gettext("Terms and conditions"),
+                "terms-and-conditions/",
+            ).__dict__,
+        ]
     )
+    powered_by_logo: str = "nisra-logo-black-en"
+    powered_by_logo_alt: str = "NISRA - Northern Ireland Statistics and Research Agency"
+    account_service_url = CENSUS_NIR_BASE_URL
+
+    def __post_init__(self) -> None:
+        self.data_layer = [{"nisra": True}]
+
+
+def generate_context(
+    theme: str,
+    base_url: str,
+    language: str,
+) -> Dict[str, Any]:
+    context_options = {
+        "business": ContextOptions,
+        "health": ContextOptions,
+        "social": ContextOptions,
+        "census": (
+            CensusContextOptionsCymraeg if language == "cy" else CensusContextOptions
+        ),
+        "default": ContextOptions,
+        "census-nisra": CensusNISRAContextOptions,
+    }
+
+    return ContextHelper(
+        theme,
+        base_url,
+        language,
+        context_options[theme](),
+    ).context
+
 
+def render_template(template: str, **kwargs: Mapping) -> str:
+    # The fallback to assigning SURVEY_TYPE to theme is only being added until
+    # business feedback on the differentiation between theme and SURVEY_TYPE.
+    theme = cookie_session.get("theme", current_app.config["SURVEY_TYPE"])
+    language = get_locale().language
+    base_url = get_base_url("theme", language)
```

Comment:
Yeah - the gap is around testing for `census-nisra` (that's the only theme check we do when getting the base url, otherwise it defaults to `CENSUS_EN_BASE_URL`

Reply:
and more specifically, testing the render_template method, which it doesn't seem is tested anywhere

Reply:
Guess we could close the gap by making the schema theme and language code like static known types instead of any arbitary string?. Prob for a future card.